codec-compression/src/main/java/io/netty/handler/codec/compression/ZstdEncoder.java

```diff
@@ -113,7 +113,7 @@ protected ByteBuf allocateBuffer(ChannelHandlerContext ctx, ByteBuf msg, boolean
         while (remaining > 0) {
             int curSize = Math.min(blockSize, remaining);
             remaining -= curSize;
-            bufferSize += Zstd.compressBound(curSize);
+            bufferSize = Math.max(bufferSize, Zstd.compressBound(curSize));
```

Comment:
Why this change ? Can you add a comment please ?

Reply:
got it... then please add a comment that compressBound returns the maximum size.